src/data/zmod/units.lean

```diff
@@ -0,0 +1,169 @@
+/-
+Copyright (c) 2023 Ashvni Narayanan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ashvni Narayanan
+-/
+import data.zmod.basic
+import topology.algebra.constructions
+
+/-!
+# Properties of units of ℤ/nℤ
+
+This file enlists some properties of units of `zmod n`. We then define a topological structure
+(the discrete topology) on `zmod n` for all `n` and consequently on the units.
+We also enlist several properties that are helpful with modular arithmetic.
+
+## Main definitions and theorems
+ * `zmod.topological_space`
+ * `cast_hom_apply'` : a version of `cast_hom_apply` where `R` is explicit
+ * `disc_top_units` : giving discrete topology to `units (zmod n)`
+
+## Tags
+zmod, units
+-/
+
+open zmod nat
+namespace zmod
+/-- Same as `zmod.cast_hom_apply` with some hypotheses being made explicit. -/
+lemma cast_hom_apply' {n : ℕ} (R : Type*) [ring R] {m : ℕ} [char_p R m]
+  (h : m ∣ n) (i : zmod n) : (cast_hom h R) i = ↑i := cast_hom_apply i
+
+lemma coe_map_of_dvd {a b : ℕ} (h : a ∣ b) {x : (zmod b)} (hx : is_unit x) :
+  is_unit (x : zmod a) := is_unit.map (cast_hom h (zmod a)).to_monoid_hom hx
+
+lemma is_unit_of_is_coprime_dvd {a b : ℕ} (h : a ∣ b) {x : ℕ} (hx : x.coprime b) :
+  is_unit (x : zmod a) :=
```

Comment:
This sounds like you're mixing two steps at once.
```suggestion
lemma is_unit_of_is_coprime {a x : ℕ} (hx : x.coprime a) :
  is_unit (x : zmod a) :=
```
and in fact that shows you can actually get an iff:
```suggestion
@[simp] lemma is_unit_iff_is_coprime {a x : ℕ} :
  is_unit (x : zmod a) \iff x.coprime a :=
```